source/gui/messageDialog.py

```diff
@@ -0,0 +1,722 @@
+		self._contentsSizer = contentsSizer
+		self._mainSizer = mainSizer
+
+		# Use SetLabelText to avoid ampersands being interpreted as accelerators.
+		text = wx.StaticText(self)
+		text.SetLabelText(message)
+		text.Wrap(self.scaleSize(self.GetSize().Width))
+		contentsSizer.addItem(text)
+		self._addContents(contentsSizer)
+
+		buttonHelper = guiHelper.ButtonHelper(wx.HORIZONTAL)
+		contentsSizer.addDialogDismissButtons(buttonHelper)
+		self._buttonHelper = buttonHelper
+		self._addButtons(buttonHelper)
+		if buttons is not None:
+			self.addButtons(*buttons)
+
+		mainSizer.Add(
+			contentsSizer.sizer,
+			border=guiHelper.BORDER_FOR_DIALOGS,
+			flag=wx.ALL,
+		)
+		# mainSizer.Fit(self)
+		self.SetSizer(mainSizer)
+		# Import late to avoid circular import
+		from gui import mainFrame
+
+		if parent == mainFrame:
+			# NVDA's main frame is not visible on screen, so centre on screen rather than on `mainFrame` to avoid the dialog appearing at the top left of the screen.
+			self.CentreOnScreen()
+		else:
+			self.CentreOnParent()
+
+	# endregion
+
+	# region Public object API
+	@singledispatchmethod
+	def addButton(
+		self,
+		id: ReturnCode,
+		/,
+		label: str,
+		*args,
+		callback: Callback_T | None = None,
+		default_focus: bool = False,
+		default_action: bool = False,
+		closes_dialog: bool = True,
+		**kwargs,
+	) -> Self:
+		"""Add a button to the dialog.
+
+		:param id: The ID to use for the button.
+			If the dialog is to be shown modally, this will also be the return value if the dialog is closed with this button.
+		:param label: Text label to show on this button.
+		:param callback: Function to call when the button is pressed, defaults to None.
+			This is most useful for dialogs that are shown modelessly.
+		:param default_focus: whether this button should receive focus when the dialog is first opened, defaults to False.
+			If multiple buttons with `default_focus=True` are added, the last one added will receive initial focus.
+		:param default_action: Whether or not this button should be the default action for the dialog, defaults to False.
+			The default action is called when the user closes the dialog with the escape key, title bar close button, system menu close item etc.
+			If multiple buttons with `default_action=True` are added, the last one added will be the default action.
+		:param closes_dialog: Whether the button should close the dialog when pressed, defaults to True.
+		:return: The updated instance for chaining.
+		"""
+		button = self._buttonHelper.addButton(self, id, label, *args, **kwargs)
+		# Get the ID from the button instance in case it was created with id=wx.ID_ANY.
+		buttonId = button.GetId()
+		self.AddMainButtonId(buttonId)
+		# Default actions that do not close the dialog do not make sense.
+		if default_action and not closes_dialog:
+			log.warning(
+				"Default actions that do not close the dialog are not supported. Forcing close_dialog to True.",
+			)
```

Comment:
should this be setting a variable too? I see it's done implicitly but perhaps it should be explicit

```suggestion
			log.warning(
				"Default actions that do not close the dialog are not supported. Forcing close_dialog to True.",
			)
			close_dialog = True
```